File: CppPractice/CppLeetcode/Leetcode48.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int topRight;
        int topLeft;
        int bottomRight;
        int bottomLeft;
        int n = matrix.size();
        for(int i=0; i<n/2;i++)
        {
            for(int j=i;j<n-i-1;j++)
            {
                topLeft = matrix[i][j];
                topRight = matrix[j][n-i-1];
                bottomRight = matrix[n-i-1][n-j-1];
                bottomLeft = matrix[n-j-1][i];

                matrix[i][j] = bottomLeft;
                matrix[j][n-i-1] =topLeft;
                matrix[n-i-1][n-j-1]= topRight;
                matrix[n-j-1][i]= bottomRight;
            }
        }
    }
};

File: JavaScriptPractice/Leetcode48.js
var rotate = function(matrix) {
    let n = matrix.length;
    let topLeft = 0;
    let topRight = 0;
    let bottomRight = 0;
    let bottomLeft = 0;
    for(let i=0;i<=n-2;i++)
    {
        for(let j =i;j<=n-2-i;j++)
        {
            topLeft = matrix[i][j];
            topRight = matrix[j][n-1-i];
            bottomRight = matrix[n-1-i][n-1-j];
            bottomLeft = matrix[n-1-j][i];

            matrix[i][j] = bottomLeft;
            matrix[j][n-1-i]= topLeft;
            matrix[n-1-i][n-1-j] = topRight;
            matrix[n-1-j][i] = bottomRight;
        }
    }
};